src/Collection.php

```diff
@@ -471,15 +447,11 @@ public function distinct(string $fieldName, $filter = [], array $options = [])
      */
     public function drop(array $options = [])
     {
-        if (! isset($options['typeMap'])) {
-            $options['typeMap'] = $this->typeMap;
-        }
-
         $server = select_server($this->manager, $options);
 
-        if (! isset($options['writeConcern']) && ! is_in_transaction($options)) {
-            $options['writeConcern'] = $this->writeConcern;
-        }
+        $options = $this->inheritWriteOptions(
+            $this->inheritTypeMap($options),
+        );
```

Comment:
In the interest of deferring `select_server()` as late as possible, this can technically be moved up to the top of the `drop()` method. Doesn't make much functional difference, though, since these wouldn't throw, but it'd be consistent with the order you use in most other methods.